osu.Game/Rulesets/Edit/Checks/CheckLowestDiffDrainTime.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit.Checks.Components;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public abstract class CheckLowestDiffDrainTime : ICheck
+    {
+        /// <summary>
+        /// Defines the minimum drain time thresholds for different difficulty ratings.
+        /// </summary>
+        protected abstract IEnumerable<(DifficultyRating rating, double thresholdMs, string name)> GetThresholds();
+
+        private const double break_time_leniency = 30 * 1000;
+
+        public CheckMetadata Metadata => new CheckMetadata(CheckCategory.Spread, "Lowest difficulty too difficult for the given drain/play time(s)");
+
+        public IEnumerable<IssueTemplate> PossibleTemplates => new IssueTemplate[]
+        {
+            new IssueTemplateTooShort(this)
+        };
+
+        public IEnumerable<Issue> Run(BeatmapVerifierContext context)
+        {
+            IReadOnlyList<IBeatmap> difficulties = context.BeatmapsetDifficulties;
+
+            if (difficulties.Count == 0)
+                yield break;
+
+            var lowestDifficulty = difficulties.OrderBy(b => b.BeatmapInfo.StarRating).First();
```

Comment:
This is subclassed and implemented by individual rulesets, so every ruleset presumably has its own separate spread rules, but nowhere in this logic do I see any actual filtering of the beatmaps being checked to the relevant ruleset? I feel like there is potential for this to be buggy and be holding difficulties to requirements they're not supposed to be held to (e.g. a catch beatmap will get a taiko drain time warning reported on it etc)